### PR TITLE
make sure private methods/props follows underscore convention

### PR DESCRIPTION
**Pull request by erwinmombay**
Closes https://github.com/ampproject/amphtml/issues/4189

@dvoytenko see lint errors in build. will fix once you have LGTM'd

### CODE REVIEW

**Comment by dvoytenko** [comment]
All errors are exactly what I'd expect and fixes are to simply add underscore, except:
- this.win - should be public in all those cases. (remove `@private`)
- Resource.resetPendingChangeSize - should be public.
- `xhr-impl.js` case - I didn't find any reason for this error. I don't see private there. Maybe it was already fixed.
- `storage-impl.js` - all of this reported methods should be public

Good thing we are doing this, since I'm personally very surprised I made most of these errors myself :)

So, one more rule I wanted to see if we can also add: do not allow `_var` naming, except for the existing cases such as `__proto__`.

**Comment by erwinmombay** [comment]
@dvoytenko just want to clarify by `_var` naming does that cover `_properties` (just to clarify since they are different on the AST)

**Comment by erwinmombay** [comment]
also, yes xhr-impl has been fixed in another PR, i just havent rebased on this one yet

**Comment by dvoytenko** [comment]
@erwinmombay Yeah, I specifically mean properties. So, the habits people have is to name private vars as `_abc`. So if they do, it's hard to notice, especially if they also forget to add `@private`. The net result of this, however, is that we loose the obfuscation on compilation.

**Comment by erwinmombay** [comment]
@dvoytenko so the thing with `_` we actually have code which says if a prop has a leading `_` it is exported. see https://github.com/ampproject/amphtml/blob/master/build-system/runner/src/org/ampproject/AmpCodingConvention.java#L83

**Comment by dvoytenko** [comment]
@erwinmombay Could you please confirm what "exported" means in this case?

**Comment by erwinmombay** [comment]
@dvoytenko by exported its treated as public and won't be mangled

**Comment by dvoytenko** [comment]
Got it. Hmm. Then it's the opposite of what trailing `_` does. I'm generally fine with the functionality. Just want to reduce the probability of unintentional use.

File: build-system/eslint-rules/enforce-private-props.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  /**
+   * @param {!Node}
+   * @return {boolean}
+   */
+  function isThisMemberExpression(node) {
+    return node.type == 'MemberExpression' &&
+        node.object.type == 'ThisExpression';
+  }
+  return {
+    'MethodDefinition:exit': function(node) {
```
Comment (jridgewell) [inline build-system/eslint-rules/enforce-private-props.js]: Why do this on exit?

Comment (erwinmombay) [inline build-system/eslint-rules/enforce-private-props.js]: no specific reason, honestly could have done this on MethodDefinition too i think. should I?

Comment (jridgewell) [inline build-system/eslint-rules/enforce-private-props.js]: Just makes it look better. 😁

File: src/service/performance-impl.js [hunk 2]
```diff
@@ -81,7 +81,7 @@ export class Performance {
     this.viewer_ = null;
 
     /** @private {?./resources-impl.Resources} */
-    this.resources = null;
+    this.resources_ = null;
```
Comment (jridgewell) [inline src/service/performance-impl.js]: How is this not breaking things?

File: build-system/eslint-rules/enforce-private-props.js (hunk 1 above)
Comment (erwinmombay) [inline build-system/eslint-rules/enforce-private-props.js]: haha done.

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 3]
```diff
@@ -111,7 +111,7 @@ export class AmpAnalytics extends AMP.BaseElement {
     /**
      * @private {JSONType}
      */
-    this.remoteConfig = {};
+    this.remoteConfig_ = {};
```
Comment (jridgewell) [inline extensions/amp-analytics/0.1/amp-analytics.js]: Same breaking change.

File: src/service/performance-impl.js (hunk 2 above)
Comment (erwinmombay) [inline src/service/performance-impl.js]: we reassign with an underscore in `coreServicesAvailable` which is where things start executing so declaration was just bad but thankfully didn't affect functionality.

File: extensions/amp-analytics/0.1/amp-analytics.js (hunk 3 above)
Comment (erwinmombay) [inline extensions/amp-analytics/0.1/amp-analytics.js]: yeah this one im looking over again

cc @avimehta

Comment (erwinmombay) [inline extensions/amp-analytics/0.1/amp-analytics.js]: @avimehta can you review this one?

Comment (avimehta) [inline extensions/amp-analytics/0.1/amp-analytics.js]: There should be a change on L251 but I don't see it in the diff.

Comment (erwinmombay) [inline extensions/amp-analytics/0.1/amp-analytics.js]: @avimehta in master right now L251 already uses an underscore https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/amp-analytics.js#L251

Comment (avimehta) [inline extensions/amp-analytics/0.1/amp-analytics.js]: ah yes. LGTM then.

**Comment by erwinmombay** [comment]
@dvoytenko @jridgewell can you do final pass?

**Comment by erwinmombay** [comment]
@dvoytenko ping, for pinal review

**Comment by dvoytenko** [comment]
LGTM

**Pull request merged**